lib/ably/AblyExceptions.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/**
+ * Generic exception for Ably classes
+ */
+class AblyException extends Exception {
+    
+    public function __construct($message, $code = 0) {
```

Comment:
FYI, an AblyException typically has both a status code and a code.  The status code often maps to a normal HTTP status code, whereas the code is a unique Ably error code that we try and share across all libraries, see https://github.com/ably/ably-common/blob/master/protocol/errors.json.  See the [Java def of a AblyException](https://github.com/ably/ably-java/blob/master/src/io/ably/types/AblyException.java#L37)